crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -88,7 +88,12 @@ impl Fulfillment {
     /// considering their signed order and the uniform clearing prices
     pub fn fee(&self) -> order::SellAmount {
         match self.fee {
-            Fee::Static => self.order.user_fee,
+            Fee::Static => {
+                // Orders with static fees are no longer used, expect for quoting purposes, when
+                // the static fee is set to 0. This is expected to be resolved with https://github.com/cowprotocol/services/issues/2543
```

Comment:
Is there anything blocking #2543 ? If not could we simply implement this now to combine the cleanups (I'm not sure there will be time scheduled for the follow up cleanup)

Reply:
Started working on it here: https://github.com/cowprotocol/services/pull/2590
Do you think #2543 should block merging this PR?